packages/contracts/contracts/FlippenZap.sol

```diff
@@ -0,0 +1,248 @@
+    /// @param _balancerPoolId The pool ID of eBTC-stETH pair in Balancer Vault
+    constructor(
+        address _borrowerOperations,
+        address _priceFeed,
+        address _cdpManager,
+        address _collToken,
+        address _ebtcToken,
+        address _balancerVault,
+        bytes32 _balancerPoolId
+    ) public {
+        require(_borrowerOperations != address(0), "FlippenZap: empty BorrowerOperations address");
+        require(_priceFeed != address(0), "FlippenZap: empty PriceFeed address");
+        require(_cdpManager != address(0), "FlippenZap: empty CdpManager address");
+        require(_collToken != address(0), "FlippenZap: empty collateral token address");
+        require(_ebtcToken != address(0), "FlippenZap: empty eBTC token address");
+        require(_balancerVault != address(0), "FlippenZap: empty Balancer vault address");
+        require(_balancerPoolId != bytes32(0), "FlippenZap: empty Balancer pool ID");
+
+        ebtcBorrowerOperations = IBorrowerOperations(_borrowerOperations);
+        ebtcPriceFeed = IPriceFeed(_priceFeed);
+        ebtcCdpManager = ICdpManager(_cdpManager);
+        ebtcCollToken = ICollateralToken(_collToken);
+        ebtcToken = IEBTCToken(_ebtcToken);
+        balancerVault = IBalancerV2Vault(_balancerVault);
+        balancerPoolId = _balancerPoolId;
+
+        ebtcCollToken.approve(_borrowerOperations, type(uint256).max);
+        ebtcToken.approve(_borrowerOperations, type(uint256).max);
+
+        ebtcCollToken.approve(_balancerVault, type(uint256).max);
+        ebtcToken.approve(_balancerVault, type(uint256).max);
+    }
+
+    /// @notice This function allows caller to create a leveraged CDP with an initial collateral.
+    /// @notice The caller has to approve this contract as a valid PositionManager via BorrowerOperations
+    /// @notice and approve this contract to spend _initialStETH amount of collateral
+    /// @param _initialStETH The initial amount of stETH used as collateral
+    /// @param _leverage The expected leverage for this long position, should be no more than MAX_REASONABLE_LEVERAGE
+    /// @return The CdpId created with owner as the caller
+    function enterLongEth(uint256 _initialStETH, uint256 _leverage) external returns (bytes32) {
+        require(_initialStETH > MIN_NET_STETH_BALANCE, "FlippenZap: initial principal too small");
+        require(_leverage <= MAX_REASONABLE_LEVERAGE, "FlippenZap: _leverage is too big");
+        require(_leverage > 0, "FlippenZap: _leverage is too small");
+
+        // under the covers:
+        // mints eBTC against stETH collateral and
+        // sells the eBTC back for stETH.
+
+        bytes32 _newCdp;
+        uint256 _collToLoan = _leverage > 1 ? (_leverage - 1) * _initialStETH : 0;
+        uint256 _expectedColl = _collToLoan + _initialStETH;
+        uint256 _expectedDebt;
+
+        // transfer initial collateral
+        ebtcCollToken.transferFrom(msg.sender, address(this), _initialStETH);
```

Comment:
`ebtcCollToken` is stETH, right?

So this needs to account for the ["1-wei corner case"](https://docs.lido.fi/guides/lido-tokens-integration-guide/#1-wei-corner-case), i.e., the `transferFrom` needs to have a before/after balance to do the accounting properly.

Reply:
Good catch ser. Fixed as suggested